packages/vite/src/plugins/paths.ts

```diff
@@ -1,16 +1,30 @@
+import { pathToFileURL } from 'node:url'
 import MagicString from 'magic-string'
+import { parseQuery, parseURL } from 'ufo'
 import type { Plugin } from 'vite'
 
 export interface RuntimePathsOptions {
   sourcemap?: boolean
 }
 
+const VITE_ASSET_RE = /__VITE_ASSET__|__VITE_PUBLIC_ASSET__/
+const JS_RE = /\.((c|m)?j|t)sx?$/
+
 export function runtimePathsPlugin (options: RuntimePathsOptions): Plugin {
   return {
     name: 'nuxt:runtime-paths-dep',
     enforce: 'post',
     transform (code, id) {
-      if (code.includes('__VITE_ASSET__') || code.includes('__VITE_PUBLIC_ASSET__')) {
+      const { pathname, search } = parseURL(decodeURIComponent(pathToFileURL(id).href))
```

Comment:
Another tiny perf: While at at moment this doesn't make a different with current version of UFO, since `search` is a getter computer on native URL, once we migrate to URL for parsing, `search` is not needed unless `pathname` check passes. In this case, not destructuring can benefit.